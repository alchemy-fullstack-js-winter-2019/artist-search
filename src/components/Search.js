import React, { Component } from 'react';
import Artists from './artists/Artists';


export default class Search extends Component {
  state = {
    // artists: [],
    query: 'Nirvana'
  };


  onSubmit(event) {
    event.preventDefault();
    console.log('hello');
  }

  handleChange = ({ target }) => {
    // const { query } = this.state;
    this.setState({ [target.name]: target.value }, () => {
      console.log('input yo');
    });
  }

  render() {
    const { query } = this.state;
    return (
      < >
      <h1>Sup Searchie</h1>
      <form onSubmit={this.onSubmit}>
        <input onChange={this.handleChange} type="text" name="query" value={query}/>
        <button>Yo</button>
      </form>
      <Artists query={query}/>
      </>
    );
  }
}

