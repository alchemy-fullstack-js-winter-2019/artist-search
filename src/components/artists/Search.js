import React, { PureComponent } from 'react';
// import PropTypes from 'prop-types';
import { getArtists } from '../../services/artistsApi';
import Artist from './Artist';

export default class Search extends PureComponent {
  state = {
    artists: [],
    searchTerm: ''
  };


 
  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    console.log('!prevent');
    getArtists(this.state.searchTerm)
      .then(response => {
        this.setState({ artists: response.results });
      });
  };

  render() {
    const artists = this.state.artists.map(artist => {
      return (
        <Artist key={artist.id} artist={artist} />
      );
    });
    return (
      <>
        <h1>Search for an Artist:</h1>
        <form onSubmit={this.handleSubmit}>
          <input name="searchTerm" value={this.state.searchTerm} onChange={this.handleChange}></input>
          <button>Search</button>
        </form>
        <ul>
          {artists}
        </ul>
      </>
    );
  }
}
