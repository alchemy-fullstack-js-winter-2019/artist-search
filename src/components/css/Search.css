@import url('https://fonts.googleapis.com/css?family=Staatliches');

.search {
  h1 {
  margin: 0;
  font-family: Staatliches;
  padding: 10px;
  text-align: center;
}
form {
  text-align: center;
  button {
    font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif
  }
}
button {
  font-family: 'Courier New', Courier, monospace;
  margin: 0 auto;
}
div {
  text-align: center;
  margin-top: 20px;
}
}
