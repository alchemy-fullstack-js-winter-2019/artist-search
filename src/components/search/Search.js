import React, { PureComponent } from 'react';
import { getArtist } from '../services/musicbrainApi';
import SearchResults from './SearchResults';

export default class Search extends PureComponent {
    state = {
        artist: '',
        results: [],
        selectedArtist: '',
        totalPages: 1,
        page: 1
    };

    handleSearch = ({ target }) => {
        this.setState({ [target.name]: target.value });
    }

    getArtistApi = (event) => {
        event.preventDefault();
        getArtist(this.state.artist, this.state.page)
            .then(res => {
                this.setState({ results: res.artists });
                this.setState({ totalPages: res.totalPages });
            });
    }

    decrementPage = () => {
        this.setState({ page: this.state.page - 1 });
    };

    incrementPage = () => {
        this.setState({ page: this.state.page + 1 });
    };

    componentDidUpdate(prevProps, prevState) {

        if(prevState.artist !== this.state.artist) {
            getArtist(this.state.artist, this.state.page)
                .then(res => {
                    this.setState({ results: res.artists });
                    this.setState({ totalPages: res.totalPages });
                });
        }
    }

    render() {
        const { results, artist, page, totalPages } = this.state;
        return (
            <>
                <h1> ARTIST SEARCH COMP</h1>
                <form onSubmit={this.getArtistApi}>
                    <label>Search for Artist
                        <input type="text" name="artist" value={artist} onChange={this.handleSearch}/>
                    </label>
                    <button >Search</button>
                </form>
                <div>
                    {page > 1 && <button onClick={this.decrementPage}>PREV</button> }
                    <span>{page}/{totalPages}</span>
                    { page < totalPages && <button onClick={this.incrementPage}>NEXT</button> }
                </div>
                <div>
                    <SearchResults
                        results={results} 
                    />
                </div>
            </>
        );
    }
}

