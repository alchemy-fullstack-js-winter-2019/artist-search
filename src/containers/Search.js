import React, { Component } from 'react';
import { ArtistsWithPaging } from '../components/artists/Artists';

export default class Search extends Component {
  state = {
    artistText: '',
    text: ''
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.setState({ artistText: this.state.text });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { artistText, text } = this.state;
    return (
    <>
      <h1>Search!!!</h1>
      <form onSubmit={this.handleSubmit}>
        <label>
          Search Name:
          <input type="text" name="text" value={text} onChange={this.handleChange}></input>
        </label>
        <button type="submit">Submit</button>
      </form>
      <ArtistsWithPaging 
        artistText={artistText}
      />
    </> 
    );
  }
}
