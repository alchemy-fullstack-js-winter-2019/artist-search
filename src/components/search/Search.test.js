import React from 'react';
import renderer from 'react-test-renderer';
import Search from './Search';

jest.mock('../services/musicbrainApi.js');

describe('Search Test', () => {
    it('matches a snapshot', () => {
        const tree = renderer.create(
            <Search />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
