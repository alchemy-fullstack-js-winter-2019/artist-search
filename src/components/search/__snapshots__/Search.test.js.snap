// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Test matches a snapshot 1`] = `
Array [
  <h1>
     ARTIST SEARCH COMP
  </h1>,
  <form
    onSubmit={[Function]}
  >
    <label>
      Search for Artist
      <input
        name="artist"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button>
      Search
    </button>
  </form>,
  <div>
    <span>
      1
      /
      1
    </span>
  </div>,
  <div>
    <ul />
  </div>,
]
`;
